### DISCUSSION TITLE
Access events object or commit history from JS actions

### DISCUSSION THREAD

**Discussion by daisy1754** [community/community#27064] [Actions]
<p>I’m trying to build a github action using JavaScript. My action wants to access list of commits included in a push.</p>
<p>I can achieve this (when not using JavaScript action) by following command:</p>
<blockquote>
<p>git rev-list <span class="math">{{ github.event.before }}..</span>{{ github.event.after }}</p>
</blockquote>
<p>Now problem is that I cannot find away to access these <code>github.event.before</code> variable from JavaScript action. It doesn’t seem like <code>@actions/github</code> npm package doesn’t expose it. Only way I could think of is asking users of my action to populate this and pass it to my action but I want to avoid it to make integration simple.</p>
<p>Is there a way for JavaScript action to access this raw event object?</p>

**Comment by joshmgross** [comment, marked as answer]
<p>:wave: Hey <a class="mention" href="/daisy1754">@daisy1754</a>,</p>
<p>The full event info can be found in a JSON file located with the <code>GITHUB_EVENT_PATH</code> environment variable. Your action should be able to load that file and get any information it needs.</p>
